src/main/java/uk/gov/hmcts/reform/et/syaapi/service/PdfMapperService.java

```diff
@@ -0,0 +1,325 @@
+import uk.gov.hmcts.et.common.model.ccd.types.RepresentedTypeC;
+import uk.gov.hmcts.et.common.model.ccd.types.RespondentSumType;
+import uk.gov.hmcts.reform.et.syaapi.constants.PdfMapperConstants;
+
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Maps Case Data attributes to fields within the PDF template.
+ * Inputs that are accepted from the ET1 form can then be mapped to
+ * the corresponding questions within the template PDF Provided
+ */
+@Service
+public class PdfMapperService {
+    private static final  Map<String, String> TITLES = Map.of(
+        "Mr", PdfMapperConstants.Q1_TITLE_MR,
+        "Mrs", PdfMapperConstants.Q1_TITLE_MRS,
+        "Miss", PdfMapperConstants.Q1_TITLE_MISS,
+        "Ms", PdfMapperConstants.Q1_TITLE_MS
+    );
+    private static final Map<String, String> TITLE_MAP = Map.of(
+        "Mr", "Mister",
+        "Mrs", "Mrs",
+        "Miss", "Mis",
+        "Ms", "Ms"
+    );
+    private static final String[] ADDRESS_PREFIX = {
+        "2.2",
+        "2.5 R2",
+        "2.7 R3",
+        "13 R4",
+        "13 R5"
+    };
+    private static final String REP_ADDRESS_PREFIX = "11.3 Representative's address:";
+    private static final String CLAIMANT_ADDRESS_PREFIX = "1.5";
+    private static final Integer MULTIPLE_RESPONDENTS = 2;
+    private static final Integer MAX_RESPONDENTS = 5;
+    private static final String YES = "Yes";
+    private static final String NO = "No";
+    private static final String[] ACAS_PREFIX = {
+        "2.3",
+        "2.6",
+        "2.8",
+        "13 R4",
+        "13 R5"
+    };
+
+    /**
+     * Maps the parameters within case data to the inputs of the PDF Template.
+     *
+     * @param caseData          the case data that is to be mapped to the inputs in the PDF Template.
+     * @return                  a Map containing the values from case data with the
+     *                          corresponding PDF input labels as a key.
+     @throws PdfMapperException should there be a problem with mapping variables
+     */
+    public Map<String, String> mapHeadersToPdf(CaseData caseData) throws PdfMapperException {
+        verifyCase(caseData);
+        Map<String, String> printFields = createHashMap();
+        printFields.put(PdfMapperConstants.TRIBUNAL_OFFICE, caseData.getManagingOffice());
+        printFields.put(PdfMapperConstants.CASE_NUMBER, caseData.getCcdID());
+        printFields.put(PdfMapperConstants.DATE_RECEIVED, caseData.getReceiptDate());
+        printFields.putAll(printPersonalDetails(caseData));
+        printFields.putAll(printRespondantDetails(caseData));
+        printFields.putAll(printEmploymentDetails(caseData));
+        printFields.putAll(printRepresentative(caseData.getRepresentativeClaimantType()));
+        return printFields;
+    }
+
+    private void verifyCase(CaseData caseData) throws PdfMapperException {
+        if (caseData == null) {
```

Comment:
If it appropriate to check caseData is null then doesn't the same logic apply to checking that
caseData.getClaimantIndType is not null and caseData.getClaimantAddressUK is not null and caseData.getRespondentCollection?

Why check that caseData is not null and then not check the properties that you depend on?

Reply:
Removed